apps/src/javalab/Playground.js

```diff
@@ -81,17 +91,36 @@ export default class Playground {
   }
 
   addClickableItem(itemData) {
-    if (this.isGameOver) {
-      // can't add new items if the game is over
-      return;
-    }
+    this.addImageHelper(itemData, true);
   }
 
   addImageItem(itemData) {
-    if (this.isGameOver) {
+    this.addImageHelper(itemData, false);
+  }
+
+  addImageHelper(itemData, isClickable) {
+    // ignore request if the game is over or if the item already exists
+    if (this.isGameOver || this.imageItemExists(itemData)) {
       // can't add new items if the game is over
       return;
     }
+    this.imageItemIds.push(itemData.id);
+    let onClick = () => {};
+    if (isClickable) {
+      onClick = () => this.handleImageClick(itemData.id);
+    }
+
+    const imageData = {
+      fileUrl: this.getUrl(itemData.filename),
+      x: itemData.x,
+      y: itemData.y,
+      height: itemData.height,
+      width: itemData.width,
+      index: itemData.index,
+      onClick: onClick,
+      type: 'image'
```

Comment:
nit: might be worth pulling this out into an exported constant since we're referring to it in other places? (i imagine we'll need one for `'text'` too right?)